src/fsharp/fsi/fsi.fs

```diff
@@ -1,6 +1,6 @@
 // Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-module Microsoft.FSharp.Compiler.Interactive.Shell
```

Comment:
Actually looking at what you have it seems you may be close to having done this as I described (sorry).